packages/sql-openapi/lib/entity-to-routes.js

```diff
@@ -108,63 +108,73 @@ async function entityPlugin (app, opts) {
+          querystring: {
+            type: 'object',
+            properties: {
+              fields: getFieldsForEntity(targetEntity, ignore)
+            }
+          },
+          response: {
+            200: {
+              type: 'array',
+              items: targetEntitySchema
+            }
           }
         },
-        response: {
-          200: {
-            type: 'array',
-            items: targetEntitySchema
-          }
+        links: {
+          200: entityLinks
         }
-      },
-      links: {
-        200: entityLinks
-      }
-    }, async function (request, reply) {
-      const ctx = { app: this, reply }
-      // IF we want to have HTTP/404 in case the entity does not exist
-      // we need to do 2 queries. One to check if the entity exists. the other to get the related entities
-      // Improvement: this could be also done with a single query with a join,
+      }, async function (request, reply) {
+        const ctx = { app: this, reply }
+        // IF we want to have HTTP/404 in case the entity does not exist
+        // we need to do 2 queries. One to check if the entity exists. the other to get the related entities
+        // Improvement: this could be also done with a single query with a join,
 
-      // check that the entity exists
-      const resEntity = await entity.count({
-        ctx,
-        where: {
-          [primaryKeyCamelcase]: {
-            eq: request.params[primaryKeyCamelcase]
+        // check that the entity exists
+        const resEntity = await entity.count({
+          ctx,
+          where: {
+            [primaryKeyCamelcase]: {
+              eq: request.params[primaryKeyCamelcase]
+            }
           }
+        })
+        if (resEntity === 0) {
+          return reply.callNotFound()
         }
-      })
-      if (resEntity === 0) {
-        return reply.callNotFound()
-      }
 
-      // get the related entities
-      const res = await targetEntity.find({
-        ctx,
-        where: {
-          [targetForeignKeyCamelcase]: {
-            eq: request.params[primaryKeyCamelcase]
-          }
-        },
-        fields: request.query.fields
+        // get the related entities
+        const res = await targetEntity.find({
+          ctx,
+          where: {
+            [targetForeignKeyCamelcase]: {
+              eq: request.params[primaryKeyCamelcase]
+            }
+          },
+          fields: request.query.fields
 
+        })
+        if (res.length === 0) {
+          // This is a query on a FK, so
+          return []
+        }
+        return res
       })
-      if (res.length === 0) {
-        // This is a query on a FK, so
-        return []
-      }
-      return res
-    })
+    } catch (error) /* istanbul ignore next */ {
+      app.log.error(error)
+      app.log.info({ routePathName, targetEntityName, targetEntitySchema, operationId })
+      throw new Error('Unable to create the route for the reverse relationship')
```

Comment:
Where is this error catched? I'm not convinced we should throw here in this case.

Reply:
For simplicity, I copy-paste the difference between `master` version:
```
/platformatic/node_modules/.pnpm/find-my-way@7.3.1/node_modules/find-my-way/index.js:301
      throw new Error(`Method '${method}' already declared for route '${path}' with constraints '${JSON.stringify(constraints)}'`)
            ^

Error: Method 'GET' already declared for route '/owners/:/field1' with constraints '{}'
    at Router._on (/platformatic/node_modules/.pnpm/find-my-way@7.3.1/node_modules/find-my-way/index.js:301:13)
    at Router.on (/platformatic/node_modules/.pnpm/find-my-way@7.3.1/node_modules/find-my-way/index.js:136:10)
    at Object.addNewRoute (/platformatic/node_modules/.pnpm/fastify@4.10.2/node_modules/fastify/lib/route.js:301:16)
    at Object.route (/platformatic/node_modules/.pnpm/fastify@4.10.2/node_modules/fastify/lib/route.js:217:19)
    at Object.prepareRoute (/platformatic/node_modules/.pnpm/fastify@4.10.2/node_modules/fastify/lib/route.js:150:18)
    at Object._get [as get] (/platformatic/node_modules/.pnpm/fastify@4.10.2/node_modules/fastify/fastify.js:249:34)
    at entityPlugin (/platformatic/packages/sql-openapi/lib/entity-to-routes.js:119:11)
    at Plugin.exec (/platformatic/node_modules/.pnpm/avvio@8.2.0/node_modules/avvio/plugin.js:130:19)
    at Boot.loadPlugin (/platformatic/node_modules/.pnpm/avvio@8.2.0/node_modules/avvio/plugin.js:272:10)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

and now:
```
[18:33:39.447] ERROR (64394): Method 'GET' already declared for route '/owners/:/field1' with constraints '{}'
    err: {
      "type": "Error",
      "message": "Method 'GET' already declared for route '/owners/:/field1' with constraints '{}'",
      "stack":
          Error: Method 'GET' already declared for route '/owners/:/field1' with constraints '{}'
              at Router._on (/platformatic/node_modules/.pnpm/find-my-way@7.3.1/node_modules/find-my-way/index.js:301:13)
              at Router.on (/platformatic/node_modules/.pnpm/find-my-way@7.3.1/node_modules/find-my-way/index.js:136:10)
              at Object.addNewRoute (/platformatic/node_modules/.pnpm/fastify@4.10.2/node_modules/fastify/lib/route.js:301:16)
              at Object.route (/platformatic/node_modules/.pnpm/fastify@4.10.2/node_modules/fastify/lib/route.js:217:19)
              at Object.prepareRoute (/platformatic/node_modules/.pnpm/fastify@4.10.2/node_modules/fastify/lib/route.js:150:18)
              at Object._get [as get] (/platformatic/node_modules/.pnpm/fastify@4.10.2/node_modules/fastify/fastify.js:249:34)
              at entityPlugin (/platformatic/packages/sql-openapi/lib/entity-to-routes.js:120:11)
              at Plugin.exec (/platformatic/node_modules/.pnpm/avvio@8.2.0/node_modules/avvio/plugin.js:130:19)
              at Boot.loadPlugin (/platformatic/node_modules/.pnpm/avvio@8.2.0/node_modules/avvio/plugin.js:272:10)
              at processTicksAndRejections (node:internal/process/task_queues:83:21)
    }
[18:33:39.448] INFO (64394):
    routePathName: "field1"
    targetEntityName: "post"
    targetEntitySchema: {
      "$ref": "Post#"
    }
    operationId: "getPostsForOwner"
/platformatic/packages/sql-openapi/lib/entity-to-routes.js:179
      throw new Error('Unable to create the route for the reverse relationship')
            ^

Error: Unable to create the route for the reverse relationship
    at entityPlugin (/platformatic/packages/sql-openapi/lib/entity-to-routes.js:179:13)
    at Plugin.exec (/platformatic/node_modules/.pnpm/avvio@8.2.0/node_modules/avvio/plugin.js:130:19)
    at Boot.loadPlugin (/platformatic/node_modules/.pnpm/avvio@8.2.0/node_modules/avvio/plugin.js:272:10)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

As you see from here, previously it was `find-my-route` to throw. I added a custom error catcher because I had really a hard time debugging it to find the root cause of the issue...

Reply:
I didn't understand this correctly, it's ok!